File: leetcode/Arrays/leetcode-0045/45.java
class Solution {
    public int jump(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int res = 0;
        int pre = 0, cur = 0;
        int i = 0;
        while (cur < nums.length - 1) {
            res++;
            pre = cur;
            for (; i <= pre; i++) {
                cur = Math.max(cur, i + nums[i]);
            }
            if (pre == cur) {
                return -1;
            }
        }
        return res;
    }
}

File: leetcode/Arrays/leetcode-0045/45.go
package arrays

func jump(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	pre, cur, i := 0, 0, 0
	res := 0
	for cur < len(nums)-1 {
		res++
		pre = cur
		for i <= pre {
			if i+nums[i] > cur {
				cur = i + nums[i]
			}
			i++
		}
		if cur == pre {
			return -1
		}
	}
	return res
}
